### PR TITLE
Add scoring and some validation for SC1

### PR DESCRIPTION
**Pull request by Jacoberts**
I want to add more to validation later, but this is most of it.

### CODE REVIEW

File: Docker/score.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  """
+  with open(submission_path, 'r') as s_file:
+    with open(goldstandard_path, 'r') as g_file:
+      scoreSC1WithFiles(s_file, g_file)
+
+
+def scoreSC1WithFiles(submission_file, goldstandard_file):
+  """Same as above, but takes file objects. Assumes files are validated."""
+  dtype = {'lab_id': str, 'drug': str}
+  submission = pandas.read_csv(submission_file, dtype=dtype)
```
Comment (thomasyu888) [inline Docker/score.py]: `pandas.read_csv` actually takes in the file path as well. So you don't necessarily need both `scoreSC1` and `scoreSC1WithFiles.`

File: Docker/score.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  goldstandard = pandas.read_csv(goldstandard_file, dtype=dtype)
+
+  # Create a single DataFrame which is indexed by (lab_id, drug) and has two
+  # columns: [auc_submission, auc_goldstandard]
+  indices = ['drug', 'lab_id']
+  submission.set_index(indices, inplace=True)
+  goldstandard.set_index(indices, inplace=True)
+  joined = submission.join(
+      goldstandard,
```
Comment (thomasyu888) [inline Docker/score.py]: Interesting, is this a join by index?

File: Docker/score.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    submission_path: path to subchallenge 1 submission file
+    goldstandard_path: path to the golden / ground truth file for subchallenge
+        1.
+  
+  Raises:
+    ValueError if submission file is invalid.
+  """
+  with open(submission_path, 'r') as s_file:
+    with open(goldstandard_path, 'r') as g_file:
+      validateSC1(s_file, g_file)
```
Comment (thomasyu888) [inline Docker/score.py]:
```suggestion
      validateSC1WithFiles(s_file, g_file)
```

File: Docker/score.py (hunk 2 above)
Comment (Jacoberts) [reply Docker/score.py]: Yep! I updated the comment to include the fact that we have a multiindex with two dimensions, lab_id and drug, and that we're joining on that index.

File: Docker/score.py (hunk 1 above)
Comment (Jacoberts) [reply Docker/score.py]: Done: I refactored to write files to a TemporaryDirectory, so that they're named (and eventually cleaned up).

**Comment by thomasyu888** [comment]
Approved! You want to merge?

**Comment by Jacoberts** [comment]
Thanks Tom!

**Pull request merged**